src/Hazelcast.Net/Serialization/ObjectDataInput.api.cs

```diff
@@ -255,7 +255,8 @@ public string[] ReadStringArray()
             if (length == BytesExtensions.SizeOfNullArray) return null;
             if (length <= 0) return Array.Empty<string>();
 
-            CheckAvailable(Position, length * BytesExtensions.SizeOfDouble);
+            //A UTF8 string is at least 1 byte.
+            CheckAvailable(Position, length * BytesExtensions.SizeOfByte);
```

Comment:
IMHO, this check is not very useful. A string will be encoded as 4 bytes length + UTF-8 payload bytes, so this check will not really make sure anything will work as expected. It might be the case that there are 3 bytes after this position, we will miss it here, but catch it in the `ReadString` calls below. 

I believe we can get rid of this check altogether, and rely on the check in the `ReadString`. This is also the way we do it in Java

Reply:
same, we can entirely remove the check

Reply:
Agreed, removed at https://github.com/hazelcast/hazelcast-csharp-client/pull/649/commits/ed9852d55214b086cba563196c6219bf27c47b71